expression/builtin_cast.go

```diff
@@ -570,6 +570,9 @@ func (b *builtinCastIntAsDurationSig) Clone() builtinFunc {
 }
 
 func (b *builtinCastIntAsDurationSig) evalDuration(row chunk.Row) (res types.Duration, isNull bool, err error) {
+	if b.tp.Decimal > types.MaxFsp {
```

Comment:
Can we put this check in `Preprocess`? that may look cleaner. Seems decimal of `time` here must be a constant, so it is safe to do check in `Preprocess`? here is the output of MySQL:
```
mysql> select * from s;
+------+------+
| a    | b    |
+------+------+
|    1 | NULL |
|    2 |    1 |
+------+------+
2 rows in set (0.00 sec)

mysql> select cast(1 as time(a)) from s;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'a)) from s' at line 1
```

Reply:
Let me have a try.